src/main/java/com/cegeka/academy/service/userChallenge/UserChallengeServiceImpl.java

```diff
@@ -26,4 +34,34 @@
         return userChallengeList.stream().map(userChallenge -> UserChallengeMapper.convertUserChallengeToUserChallengeDTO(userChallenge)).collect(Collectors.toList());
 
     }
+
+    @Override
+    public UserChallenge rateUser(UserChallengeDTO userChallengeDTO, Long ownerId)
+            throws WrongOwnerException, NoSuchElementException {
+
+        long userId = userChallengeDTO.getUser().getId();
+        long challengeId = userChallengeDTO.getChallenge().getId();
+        long invitationId = userChallengeDTO.getInvitation().getId();
+        UserChallenge userChallenge = userChallengeRepository
+                .findByUserIdAndChallengeIdAndInvitationId(userId, challengeId, invitationId);
+
+        System.out.println(userChallenge);
+
+        if (userChallenge != null) {
+            userChallenge.setPoints(userChallengeDTO.getPoints());
+
+            Optional<Challenge> challengeOptional = challengeRepository.findById(userChallenge.getChallenge().getId());
+
```

Comment:
Acesta linie nu mai are sens. De ce sa te duci la baza de date cand tu ai preluat userChallenge de acolo. Este clar ca userChallenge.getChallenge iti ofera un Challenge stocat in baza de date. Nu mai este nevoie sa apelezi findBy ca sa gasesti Challenge-ul. Il ai la dispozitie prin preluare UserChallenge-ului. Te rog sa citest despre mapping-urile din JPA si cum functioneaza.